2022.2.2+ (#2)
DevExpress-Examples/wpf-scheduler-load-data-on-demand: Readme.md

[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
<!-- default badges end -->

# WPF Scheduler - How to load data on demand

> This example loads appointments that fall into the specified date interval. The solution consists of three parts.

Reviewer: ```suggestion
This example loads appointments that fall into the specified date interval. The solution consists of three parts:
```

Author: Accepted.

# How to load Scheduler Control data on demand

> This example loads data in the Scheduler only for the selected date interval. The example consists of three parts.

Reviewer 2: ```suggestion
This Scheduler control example only loads appointments that fall into the specified date interval. The solution consists of three parts.
```

> * _CommonDbContext_. This solution uses a single `DbContext` for the application and has own SchedulingViewModel.

Reviewer 2: ```suggestion
* _CommonDbContext_. This project implements a `SchedulingViewModel` that uses a single `DbContext` for the application.
```

Author: Accepted.

> * _DbContextPerRequest_. This solution follows the One `DbContext` per request approach and has own SchedulingViewModel.

Reviewer 2: ```suggestion
* _DbContextPerRequest_. This project implements a `SchedulingViewModel` that creates a new `DbContext` per each request.
```

> * _Shared_. The shared project contains the data and the view, which are the same for both solutions. The events are mapped to the view model’s handlers.

Reviewer 2: I would move this item to the first position within the list. This is something introductory in nature - you may want to review the code here, but this is not the project you would use. 

```suggestion
* _Shared_. This shared project contains the data and the view used by other projects. The view maps events to view model’s handlers.
```

## Files to Review

> - [SchedulingViewModel.cs](./CS/CommonDbContext/ViewModels/SchedulingViewModel.cs) ([SchedulingViewModel.vb](./VB/CommonDbContext/ViewModels/SchedulingViewModel.vb))
> - [SchedulingViewModel.cs](./CS/DbContextPerRequest/ViewModels/SchedulingViewModel.cs) ([SchedulingViewModel.vb](./VB/DbContextPerRequest/ViewModels/SchedulingViewModel.vb))
> - [SchedulingView.xaml](./CS/Shared/Views/SchedulingView.xaml) ([SchedulingView.xaml](./VB/Shared/Views/SchedulingView.xaml))
> - [SchedulingView.xaml.cs](./CS/Shared/Views/SchedulingView.xaml.cs) ([SchedulingView.xaml.vb](./VB/Shared/Views/SchedulingView.xaml.vb))

Reviewer 2: Let's break down the list by project - otherwise it's hard to see what belong where. 

```suggestion
_Shared_ project:

- [SchedulingView.xaml](./CS/Shared/Views/SchedulingView.xaml) ([SchedulingView.xaml](./VB/Shared/Views/SchedulingView.xaml))
- [SchedulingView.xaml.cs](./CS/Shared/Views/SchedulingView.xaml.cs) ([SchedulingView.xaml.vb](./VB/Shared/Views/SchedulingView.xaml.vb))

_CommonDbContext_ project:

- [SchedulingViewModel.cs](./CS/CommonDbContext/ViewModels/SchedulingViewModel.cs) ([SchedulingViewModel.vb](./VB/CommonDbContext/ViewModels/SchedulingViewModel.vb))

_DbContextPerRequest_ project: 

- [SchedulingViewModel.cs](./CS/DbContextPerRequest/ViewModels/SchedulingViewModel.cs) ([SchedulingViewModel.vb](./VB/DbContextPerRequest/ViewModels/SchedulingViewModel.vb))
```
